### PR TITLE
[toml11] Add a new port

### PR DESCRIPTION
**Pull request by niyue**
**Describe the pull request**
Add a new port called toml11, https://github.com/ToruNiina/toml11
It is a header only library for parsing TOML

- Issue related
https://github.com/microsoft/vcpkg/issues/9935

- Which triplets are supported/not supported? Have you updated the CI baseline?
macOS/Windows/Linux
CI baseline not updated

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

File: ports/toml11/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
+include(vcpkg_common_functions)
```
Comment (PhoebeHui) [inline ports/toml11/portfile.cmake]: Could you please remove it? this is deprecated.

File: ports/toml11/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: toml11
+Version: v3.3.0
```
Comment (PhoebeHui) [inline ports/toml11/CONTROL]: Could you remove 'v' from 'v3.3.0'?

File: ports/toml11/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO ToruNiina/toml11
+	
+	REF v3.3.0
+	SHA512 3de018972bd2d9da193ca3d4746f1a77a120d563dbfb0e8f72800a0b01e24fb95fe83e4bd1a93d4bc10b3a603e7e63dd9478079bd0a31692d8412cbcf4860c37
+
```
Comment (PhoebeHui) [inline ports/toml11/portfile.cmake]: Could you remove the blank lines?

File: ports/toml11/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+	SHA512 3de018972bd2d9da193ca3d4746f1a77a120d563dbfb0e8f72800a0b01e24fb95fe83e4bd1a93d4bc10b3a603e7e63dd9478079bd0a31692d8412cbcf4860c37
+	
+    HEAD_REF master
+)
+
+file(INSTALL ${SOURCE_PATH}/toml.hpp DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+file(INSTALL ${SOURCE_PATH}/toml DESTINATION ${CURRENT_PACKAGES_DIR}/include FILES_MATCHING PATTERN "*.hpp")
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/toml11 RENAME copyright)
```
Comment (PhoebeHui) [inline ports/toml11/portfile.cmake]: We usually use 'file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright) instead.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@niyue, thanks for your contribution!

File: ports/toml11/portfile.cmake (hunk 1 above)
Comment (niyue) [reply ports/toml11/portfile.cmake]: Removed.

File: ports/toml11/portfile.cmake (hunk 3 above)
Comment (niyue) [reply ports/toml11/portfile.cmake]: Two blank lines are removed.

File: ports/toml11/portfile.cmake (hunk 4 above)
Comment (niyue) [reply ports/toml11/portfile.cmake]: Replaced port name with `${PORT}`

File: ports/toml11/CONTROL (hunk 2 above)
Comment (niyue) [reply ports/toml11/CONTROL]: The leading `v` is removed.

**Comment by niyue** [comment]
@PhoebeHui Thanks for your comments, the PR is updated accordingly and please have a review again.

**Review by PhoebeHui — APPROVED** [review]
Thanks for the updates!

**Comment by ras0219-msft** [comment]
LGTM, thanks for the PR!

**Pull request merged**